sdks/java/io/hadoop-common/src/main/java/org/apache/beam/sdk/io/hadoop/SerializableConfiguration.java

```diff
@@ -49,34 +52,40 @@ public SerializableConfiguration(Configuration conf) {
     if (conf == null) {
       throw new NullPointerException("Configuration must not be null.");
     }
+    this.confMutated = true;
     this.conf = conf;
   }
 
   public Configuration get() {
-    if (serializationCache != null) {
-      serializationCache = null;
-    }
+    // get() call returns the original conf, which is mutable by caller
+    confMutated = true;
     return conf;
   }
 
   @Override
   public void writeExternal(ObjectOutput out) throws IOException {
-    if (serializationCache == null) {
-      ByteArrayOutputStream baos = new ByteArrayOutputStream(512);
-      try (DataOutputStream dos = new DataOutputStream(baos)) {
-        conf.write(dos);
-        serializationCache = baos.toByteArray();
+    String canonicalName;
+    synchronized (this) {
+      canonicalName = conf.getClass().getCanonicalName();
+      if (confMutated) {
```

Comment:
```suggestion
      if (confMutated || serializationCache == null) {
```

Since `confMutated` is transient and will deserialize to `false`

Reply:
Thanks for pointing out